gateways/adyen/request_builder.go

```diff
@@ -92,7 +93,12 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest, merchantAccount s
 	}
 
 	level3 := authRequest.Level3Data
-	if level3 != nil {
+	if authRequest.Options[googlePayTokenOption] != nil {
+		googlePayData := map[string]interface{}{
+			"payment.token": authRequest.Options[googlePayTokenOption].(string),
+		}
+		request.AdditionalData = googlePayData
+	} else if level3 != nil {
```

Comment:
I think with this logic, we will skip building the level3 data if google pay is present. Lets adjust to do both

Reply:
We don't want to build the l3 data for Google Pay

Reply:
we should send both, its up to adyen to decide if the l2l3 data applies to the transaction

Reply:
Addressed